spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationBootstrapConfiguration.java

```diff
@@ -67,31 +66,50 @@ public CloseableHttpClient closeableHttpClient() {
 
     @Bean
     public AppConfigurationPropertySourceLocator sourceLocator(AppConfigurationProperties properties,
-            AppConfigurationProviderProperties appProperties, ClientStore clients, ApplicationContext context) {
+            AppConfigurationProviderProperties appProperties, ClientStore clients, ApplicationContext context,
+            Optional<KeyVaultCredentialProvider> keyVaultCredentialProviderOptional,
+            Optional<SecretClientBuilderSetup> keyVaultClientProviderOptional) {
+        
         KeyVaultCredentialProvider keyVaultCredentialProvider = null;
-        try {
-            keyVaultCredentialProvider = context.getBean(KeyVaultCredentialProvider.class);
-        } catch (NoUniqueBeanDefinitionException e) {
-            throw new RuntimeException("Failed to find unique KeyVaultCredentialProvider Bean for authentication.", e);
-        } catch (NoSuchBeanDefinitionException e) {
+        SecretClientBuilderSetup keyVaultClientProvider = null;
+        
+        if (!keyVaultCredentialProviderOptional.isPresent()) {
             LOGGER.debug("No KeyVaultCredentialProvider found.");
+        } else {
+           keyVaultCredentialProvider = keyVaultCredentialProviderOptional.get(); 
         }
+
+        if (!keyVaultClientProviderOptional.isPresent()) {
+            LOGGER.debug("No KeyVaultCredentialProvider found.");
+        } else {
+           keyVaultClientProvider = keyVaultClientProviderOptional.get(); 
+        }
```

Comment:
To read the affirmative statement it's used to be easier than the negative statement, so what do you think about revert this if/else condition? (Sugar syntax but it's helpful for a more natural reading )
```suggestion
        if (keyVaultClientProviderOptional.isPresent()) {
           keyVaultClientProvider = keyVaultClientProviderOptional.get(); 
        } else {
           LOGGER.debug("No KeyVaultCredentialProvider found.");
        }
```